### PR TITLE
Build system refactoring

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: build/GraphBLAS/Makefile [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+DRY_RUN:=
+endif
+
+#----------------------------------------------------------------------------------------------
+
+ROOT=../..
+
+ifeq ($(wildcard $(ROOT)/deps/readies/*),)
+___:=$(shell git submodule update --init --recursive &> /dev/null)
+endif
```
Comment (AviAvni) [inline build/GraphBLAS/Makefile:14]: we can remove this

File: build/xxHash/Makefile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+TARGET=$(BINDIR)/libxxhash.a
+
+#----------------------------------------------------------------------------------------------
+
+MK_CUSTOM_CLEAN=1
+
+include $(MK)/defs
+
+CC=gcc
```
Comment (AviAvni) [inline build/xxHash/Makefile:26]: why need to set this to gcc?

File: src/common.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#pragma once
+
+#if defined(DEBUG) || !defined(NDEBUG)
```
Comment (AviAvni) [inline src/common.h:4]: why the if is different from the src/debug.c

**Review by AviAvni — CHANGES_REQUESTED** [review]

File: build/xxHash/Makefile (hunk 2 above)
Comment (rafie) [reply build/xxHash/Makefile:26]: good catch. not needed and will break with CLANG=1.

File: src/common.h (hunk 3 above)
Comment (rafie) [reply src/common.h:4]: true, should be similar.

Comment (rafie) [reply src/common.h:4]: also note that `common.h` is being #include-ed into all source files in `RedisGraph/src` via `-include src/common.h`. This is good for making global definitions that should reach all compilation units.

**Comment by swilly22** [comment]
@rafie sorry for catching up on this work so late, but I prefer we do not touch Makefiles of our dependencies.

**Comment by AviAvni** [comment]
> @rafie sorry for catching up on this work so late, but I prefer we do not touch Makefiles of our dependencies.

it is not changes to the GraphBLAS makefiles it is new makefiles

**Pull request closed without merging**